src/grammar-registry.coffee

```diff
@@ -57,15 +55,17 @@ class GrammarRegistry extends FirstMate.GrammarRegistry
     return -1 unless filePath
     filePath = filePath.replace(/\\/g, '/') if process.platform is 'win32'
 
-    pathComponents = filePath.toLowerCase().split(PathSplitRegex)
+    pathComponents = filePath.toLowerCase().split('/')
+    pathComponents = pathComponents.concat(pathComponents.pop().split('.'))
```

Comment:
Now that we're not using the regex anymore, maybe we could get rid of the conditional on line 56 that replaces `\` with `/` on windows.

```coffee
pathComponents = filePath.split(path.sep)
pathComponents.push(pathComponents.pop().split('.')...)
```